src/couch/src/couch_db.erl

```diff
@@ -608,6 +609,25 @@ get_db_info(Db) ->
     ],
     {ok, InfoList}.
 
+get_design_doc(#db{name = <<"shards/", _/binary>> = ShardDbName}, DDocId0) ->
+    DDocId = case DDocId0 of
```

Comment:
Maybe abstract this cut'n'paste bit in a helper, something like `normalize_ddoc_id/1`?
```erlang
normalize_ddoc_id(<<"_design/", _/binary>> = DDocId) ->
    DDocId;
normalize_ddoc_id(DDocId) ->
   <<"_design/", DDocId>>.
```

Reply:
Yeah, I wasn't super happy with that bit. I guess adding a couch_util function might be better.

Reply:
Fixed.